.changeset/old-rats-remember.md

```diff
@@ -32,3 +32,107 @@ ArrayFormatter
+- rename `undefined` to `Undefined`
+- rename `void` to `Void`
+- rename `null` to `Null`
+- rename `never` to `Never`
+- rename `unknown` to `Unknown`
+- rename `any` to `Any`
+- rename `string` to `String`
+- rename `number` to `Number`
+- rename `boolean` to `Boolean`
+- rename `bigintFromSelf` to `BigIntFromSelf`
+- rename `symbolFromSelf` to `SymbolFromSelf`
+- rename `object` to `Object`
+- rename `union` to `Union`
+- rename `nullable` to `Nullable`
+- rename `orUndefined` to `OrUndefined`
+- rename `nullish` to `Nullish`
+- rename `keyof` to `Keyof`
+- rename `optionalElement` to `OptionalElement`
+- rename `tuple` to `Tuple`
+- rename `array` to `Array`
+- rename `nonEmptyArray` to `NonEmptyArray`
+- rename `struct` to `Struct`
+- rename `record` to `Record`
+- rename `suspend` to `Suspend`
+- rename `symbol` to `Symbol`
+- rename `bigint` to `BigInt`
+- rename `Bigint` to `BigInt`
+- rename `optionFromSelf` to `OptionFromSelf`
+- rename `option` to `Option`
+- rename `optionFromNullable` to `OptionFromNullable`
+- rename `optionFromNullish` to `OptionFromNullish`
+- rename `optionFromOrUndefined` to `OptionFromOrUndefined`
+- rename `eitherFromSelf` to `EitherFromSelf`
+  - rename `right`, `left` arguments to `Right`, `Left`
+- rename `either` to `Either`
+  - rename `right`, `left` arguments to `Right`, `Left`
+- rename `eitherFromUnion` to `EitherFromUnion`
+  - rename `right`, `left` arguments to `Right`, `Left`
+- rename `readonlyMapFromSelf` to `ReadonlyMapFromSelf`
+  - rename `key`, `value` arguments to `Key`, `Value`
+- rename `mapFromSelf` to `MapFromSelf`
+  - rename `key`, `value` arguments to `Key`, `Value`
+- rename `readonlyMap` to `ReadonlyMap`
+- rename `map` to `Map`
+- rename `readonlySetFromSelf` to `ReadonlySetFromSelf`
+- rename `setFromSelf` to `SetFromSelf`
+- rename `readonlySet` to `ReadonlySet`
+- rename `set` to `Set`
+- rename `chunkFromSelf` to `ChunkFromSelf`
+- rename `chunk` to `Chunk`
+- rename `dataFromSelf` to `DataFromSelf`
+- rename `data` to `Data`
+- rename `causeFromSelf` to `CauseFromSelf`
+  - rename `error`, `defect` arguments to `Error`, `Defect`
+- rename `causeDefectUnknown` to `CauseDefectUnknown`
+- rename `cause` to `Cause`
+  - rename `error`, `defect` arguments to `Error`, `Defect`
+- rename `exitFromSelf` to `ExitFromSelf`
+  - rename `failure`, `success`, `defect` arguments to `Failure`, `Success`, `Defect`
+- rename `exit` to `Exit`
+  - rename `failure`, `success`, `defect` arguments to `Failure`, `Success`, `Defect`
+- rename `hashSetFromSelf` to `HashSetFromSelf`
+- rename `hashSet` to `HashSet`
+- rename `hashMapFromSelf` to `HashMapFromSelf`
+  - rename `key`, `value` arguments to `Key`, `Value`
+- rename `hashMap` to `HashMap`
+  - rename `key`, `value` arguments to `Key`, `Value`
+- rename `listFromSelf` to `ListFromSelf`
+- rename `list` to `List`
+- rename `sortedSetFromSelf` to `SortedSetFromSelf`
+- rename `sortedSet` to `SortedSet`
+
+?
+
+- `encodedSchema`
```

Comment:
Just my 2 cents, PascalCase for all of the data types, to mirror TypeScript types conventions is a win and they are what I've always wanted as they will never shadow any runtime variables by most JS/TS conventions being camelCase. They're also nouns in my mind, they are definitely "things", like `User` or `Profile`.

I think the verbs listed below continue to make sense as camelCase as they represent "doing" something. If we were to follow my logic though, some of this list below aren't really verbs, so prefixing them may make sense to make them verbs to represent the active transformations or getters e.g. `encodedSchema` might make more sense as `getEncodedSchema` or similar. `partial` isn't really a verb either, so maybe `asPartial` or `toPartial`.

I'll try to come back and make more explicit suggestions when I have more time to write something if it'd be helpful

Reply:
Prefixing everything with `to`, `get` and `as` would be weird imho. Schemas are fundamentally a declarative API and `get` doesn't work well within that context imho.

Reply:
Note: I'll go through and make changes following these simple rules:
- (constructors) everything that builds a brand new Schema -> PascalCase (TypeSchema, EncodedSchema, Partial, Omit, Tuple)
- (modifiers) CamelCase (filer, annotations, message, asSchema)
- (PropertySignature constructors) CamelCase (optional, optionalToRequired, propertySignature, fromKey)

so at least we have a more consistent foundation to reason about

Reply:
@fubhy FWIW I believe declarative APIs should read as close to human language as possible such that the user doesn't need to understand the inner-workings to know what it means, a mirror to the technical goals of a declarative API - to hide the inner workings, to focus on "what" and not "how". The API should convey all of its meaning using natural language, something we all share by default. You and I don't communicate what we do or did without a verb, why should our code be any different?

Reply:
```ts
type PartialPerson = Partial<Person>
```

This is plain TS, it isn't `type PartialPerson = GetPartial<Person>`

The distinction in this case is that some types are finished and some other types require parameters to be provided

Reply:
I'd use your own words from the corresponding issue here - https://github.com/Effect-TS/effect/issues/2417#issuecomment-2025964530
> not sure I'd follow what TS does in this case, it doesn't really make sense it seems more a historical inconsistency

Reply:
That's not just typescript it's standard of type constructors, `Option<A>` not `GetOption<A>`

Reply:
There's a difference in a type constructor and a transformation though. In case of `Option<A>` is placed _within_ another type with semantics it should be `Option<A>`, like a value-level constructor function/class would. It created a "thing", a noun with semantics of its own. 

`Partial<A>` takes some `A` and computes an output, like a value-level function would. It's literally a for-loop at the type-level.

Reply:
@TylorS 

> the technical goals of a declarative API - to hide the inner workings, to focus on "what" and not "how".

and

> `Partial<A>` takes some `A` and computes an output, like a value-level function would. It's literally a for-loop at the type-level. [therefore it should be `getPartial`]

are contradictory statements 🤔 
_What_ it is, is a `Partial<User>`, _how_ it does it is by computing it and `GetPartial<User>` (read: get partial from user)

I agree with the overall sentiment, and there is an obvious distinction between `Partial<A>` and an `Option<A>`, but I believe that the difference is heuristic and not semantic

Reply:
"What" I want is to compute a partial from some type, "how" is through mapped types, but as a user I only care "what" I'm getting back, and not the syntax or choices made to compute the "what". In the case of `type PartialUser = Partial<User>`, which it is to me at least, is `PartialUser`, what was returned. `Partial<User>` itself is a description of a specific computation.

I think this is entirely symmetrical with how declarative functions work. We just use `Effect.andThen`, but we shouldn't need to know how it works beneath the hood. "What" we want to do is chain together a sequence of Effects, "how" it does it is quite a few steps starting with checks on all the options, and then building the correct sequences of Effects.